addons/loyalty_delivery/tests/test_loyalty_delivery.py

```diff
@@ -71,17 +72,105 @@ def test_delivery_cost_gift_card(self):
                 'price_unit': 120.00,
             })]
         })
-        self._apply_promo_code(sale_normal_delivery_charges, gift_card.code)
-        sale_normal_delivery_charges.action_confirm()
+        self._apply_promo_code(order, gift_card.code)
+        order.action_confirm()
+
+        delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context({
+            'default_order_id': order.id,
+            'default_carrier_id': self.free_delivery.id,
+        }))
```

Comment:
Since we are making changes anyway, let's nitpick a bit =]
```suggestion
        delivery_wizard = Form(self.env['choose.delivery.carrier'].with_context(
            {'default_order_id': order.id, 'default_carrier_id': self.free_delivery.id}
        ))
```